core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -358,44 +360,55 @@ public Object visitGetAllCommand(InvocationContext ctx, GetAllCommand command) t
          }
          return invokeNext(ctx, command);
       }
+      GetAllSuccessHandler getAllSuccessHandler = new GetAllSuccessHandler(command);
```

Comment:
You got me wrong; When you execute functional modification in a transaction, by default you don't fetch the whole entry into originator's context, just the version for WSC. However, currently the xsite is implemented in such a way that the whole transaction is replicated xsite by the originator. And since the xsite may not have the previous value, we can't just replay the func modification in there - we need to send the whole updated value. Therefore, if xsite is on, we need to fetch the modified entry in order to have the updated value on the originator.
Since some commands work on multiple entries, I need some kind of `remoteGetAll` to get hold of those values. Note that at some places the commands use just multiple `remoteGet`s but that's a bug, in fact https://issues.jboss.org/browse/ISPN-7889